action.yml

```diff
@@ -48,7 +48,6 @@ runs:
         echo "JAMCR_PASS=$JAMCR_PASS" >> $GITHUB_ENV
 
     - name: Log into Jamsocket's Docker registry
-      if: github.event_name != 'pull_request'
```

Comment:
I think this is a security thing so the secret doesn't get leaked, cc @paulgb 

I see this in plane as well https://github.com/jamsocket/plane/blob/main/.github/workflows/build-image.yml#L39

Reply:
It's not so much about leaking secrets, but about not pushing before a pull request is merged.

But in the build and push step, it should be accompanied by

```yaml
push: ${{ github.event_name != 'pull_request' }}
```

instead of

```yaml
push: true
```

Reply:
I think we can recommend folks add `if: github.event_name != 'pull_request'` when they're setting up their workflows, but I don't think we should make that decision for them _inside_ the github action. Customers may want branch previews where they want to build/push on pull_requests, tagging with the branch name. But the github action as is doesn't allow for that.

Reply:
That makes sense, but does it mean that we need to expose `push` as a variable on the action?

Reply:
Oh, you mean if the user wants to just build but not push (to confirm that building succeeds)? I think we could add that

Reply:
Let me know what you think about that latest commit